contracts/src/v0.8/feeds/test/DualAggregator.t.sol

```diff
@@ -633,58 +643,267 @@ contract LatestConfigDigestAndEpoch is TransmittedDualAggregatorBaseTest {
+    vm.startPrank(address(102));
+    assertEq(harness.latestTimestamp(), 10);
   }
 }
-contract LatestRound is TransmittedDualAggregatorBaseTest {
-  function test_ReturnsLatestRound() public view {
-    assertEq(aggregator.latestRound(), 1);
+
+// latestRound(): test all the paths
+contract LatestRound is ConfiguredDualAggregatorBaseTest {
+  function setUp() public override {
+    super.setUp();
+
+    harness.injectTransmissions();
+    harness.setLatestRoundIds(6, 2);
+    vm.warp(30);
+  }
+
+  // test return the latest primary round id
+  function test_ReturnsLatestPrimaryRoundId() public {
+    vm.warp(31);
+    assertEq(harness.latestRound(), 6);
+  }
+
+  // test latest primary round in the same block, return previous one
+  function test_ReturnsPreviousPrimaryRoundId() public view {
+    assertEq(harness.latestRound(), 5);
+  }
+
+  // test return the latest secondary round id
+  function test_ReturnsLatestSecondaryRoundId() public {
+    harness.setCutoffTime(20);
+
+    vm.startPrank(address(102));
+    assertEq(harness.latestRound(), 2);
+  }
+
+  // test return the secondary round id synced with primary rounds
+  function test_ReturnsSyncedRoundId() public {
+    harness.setCutoffTime(9);
+
+    vm.startPrank(address(102));
+    assertEq(harness.latestRound(), 4);
   }
 }
-contract GetAnswer is TransmittedDualAggregatorBaseTest {
-  function test_ReturnsCorrectAnswer() public view {
-    assertEq(aggregator.getAnswer(1), 1);
+
+// getAnswer(): test primary and secondary caller
+contract GetAnswer is ConfiguredDualAggregatorBaseTest {
+  function setUp() public override {
+    super.setUp();
+
+    harness.injectTransmissions();
+    harness.setLatestRoundIds(6, 2);
+    vm.warp(30);
+  }
+
+  // test return primary answer
+  function test_ReturnsPrimaryGetAnswer() public {
+    vm.warp(31);
+    assertEq(harness.getAnswer(6), 15);
+  }
+
+  // test return primary answer, not allowed
+  function test_ReturnsPrimaryGetAnswerNotAllowed() public view {
+    assertEq(harness.getAnswer(6), 0);
+  }
+
+  // test return secondary answer
+  function test_ReturnsSecondaryGetAnswer() public {
+    vm.startPrank(address(102));
+    assertEq(harness.getAnswer(2), 11);
+  }
+
+  // test return secondary answer, not allowed
+  function test_ReturnsSecondaryGetAnswerNotAllowed() public {
+    harness.setCutoffTime(20);
+
+    vm.startPrank(address(102));
+    assertEq(harness.getAnswer(3), 0);
   }
 }
-contract GetTimestamp is TransmittedDualAggregatorBaseTest {
-  function test_ReturnsCorrectTimestamp() public view {
-    assertEq(aggregator.getTimestamp(1), 1);
+
+// getTimestamp(): test primary and secondary caller
+contract GetTimestamp is ConfiguredDualAggregatorBaseTest {
+  function setUp() public override {
+    super.setUp();
+
+    harness.injectTransmissions();
+    harness.setLatestRoundIds(6, 2);
+    vm.warp(30);
+  }
+
+  // test return primary timestamp
+  function test_ReturnsPrimaryGetTimestamp() public {
+    vm.warp(31);
```

Comment:
going off of my above comment you can then do `vm.warp(startingBlock+1)`

Reply:
The warp defines the timestamp, it has to be 31 because of the defined transmissions.